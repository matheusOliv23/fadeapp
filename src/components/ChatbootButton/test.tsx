import { fireEvent, render, screen, waitFor } from '@testing-library/react'

import { ChatbootButton } from '.'

const { getByRole, getByText, queryByRole } = screen

describe('<ChatbootButton />', () => {
  it('Should render the ChatbootButton', () => {
    render(<ChatbootButton />)

    const text = getByText('Chat')
    const button = getByRole('button')
    const modal = queryByRole('dialog')

    expect(text).toBeInTheDocument()
    expect(button).toBeInTheDocument()
    expect(modal).not.toBeInTheDocument()
  })

  it('Should open modal on button click', async () => {
    render(<ChatbootButton />)

    const button = getByRole('button')

    fireEvent.click(button)

    await waitFor(() => {
      const modal = getByRole('dialog')

      expect(modal).toBeInTheDocument()
    })
  })

  it('should match snapshot', () => {
    const { container } = render(<ChatbootButton />)

    expect(container).toMatchSnapshot()
  })
})
