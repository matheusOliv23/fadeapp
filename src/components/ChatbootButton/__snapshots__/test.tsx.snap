// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChatbootButton /> should match snapshot 1`] = `
<div>
  <div>
    <button
      class="text-secondary
  fixed
  cursor-pointer
  bg-primary
  flex
  items-center
  justify-center
  rounded-full
  w-[3.125rem]
  h-[3.125rem]
  lg:w-[3.75rem]
  lg:h-[3.75rem]
  bottom-[5.375rem]
  right-[1.563rem]
  lg:bottom-[3.563rem]
  lg:right-[3.563rem]
  hover:bg-transparent
  hover:text-primary
  hover:border-primary
  hover:border-[0.063rem]
  transition ease-in-out
  focus:outline-none
  focus:ring
  focus:ring-0
  focus:ring-primary
  active:bg-transparent
  "
    >
      <span
        class="
  text-white
  text-lg
  lg:text-xl
  fixed
  right-[2rem]
  bottom-[9rem]
  lg:bottom-[8rem]
  lg:right-[4.3rem]
  leading-6
  font-medium
"
      >
        Chat
      </span>
      <span />
    </button>
  </div>
</div>
`;
